src/DurableTask.Netherite/Connections/SimpleCredentialResolver.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace DurableTask.Netherite
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using Azure.Core;
+    using Microsoft.Extensions.Logging.Abstractions;
+
+    /// <summary>
+    /// Resolves connections using a token credential, a storage account name, and an eventhubs namespace name.
+    /// </summary>
+    public class SimpleCredentialResolver : ConnectionResolver
```

Comment:
Yes, that seems like a good way to do it. Will adjust.

Reply:
Makes sense. Though I prefer to keep the actual classes in their own file. Just made them internal and added public factory methods to the base class.

Reply:
That approach is completely fine as well. I keep them in their own file as well - but I make the outer class `partial` and move the nested class to its own file. So, it would be `ConnectionResolver.SimpleCredentialResolver.cs`.

But not need for that if you already made it `internal`.